### PR TITLE
Fixes gh-3697

### PR DESCRIPTION
**Pull request by andrei-ivanov**
Fixes gh-3697.

I have signed and agree to the terms of the SpringSource Individual Contributor License Agreement.

### CODE REVIEW

File: web/src/test/java/org/springframework/security/web/authentication/switchuser/SwitchUserFilterTests.java [hunk 1]
```diff
@@ -396,12 +403,14 @@ public void modificationOfAuthoritiesWorks() {
 	// SEC-1763
 	@Test
 	public void nestedSwitchesAreNotAllowed() throws Exception {
```
Comment (rwinch) [inline web/src/test/java/org/springframework/security/web/authentication/switchuser/SwitchUserFilterTests.java]: Let's leave this test alone and add a new test for the new functionality.

File: web/src/main/java/org/springframework/security/web/authentication/switchuser/SwitchUserFilter.java [hunk 2]
```diff
@@ -528,6 +529,15 @@ public void setUsernameParameter(String usernameParameter) {
 	}
 
 	/**
+	 * Allows the role of the switchAuthority to be customized.
+	 * 
+	 * @param switchAuthorityRole the role name. Defaults to {@link #ROLE_PREVIOUS_ADMINISTRATOR}
+	 */
+	public void setSwitchAuthorityRole(String switchAuthorityRole) {
```
Comment (rwinch) [inline web/src/main/java/org/springframework/security/web/authentication/switchuser/SwitchUserFilter.java]: Can you add an Assert.notNull check here? Please also ensure to add a test for this.

**Comment by rwinch** [comment]
Thanks for the fast PR! I have added feedback to the code.

File: web/src/main/java/org/springframework/security/web/authentication/switchuser/SwitchUserFilter.java (hunk 2 above)
Comment (andrei-ivanov) [inline web/src/main/java/org/springframework/security/web/authentication/switchuser/SwitchUserFilter.java]: Done. I was just following the example of the other setters.

**Comment by andrei-ivanov** [comment]
I've updated the PR.

**Comment by rwinch** [comment]
Thanks for the PR! this is now merged into master via 9008a7a

File: web/src/main/java/org/springframework/security/web/authentication/switchuser/SwitchUserFilter.java (hunk 2 above)
Comment (rwinch) [inline web/src/main/java/org/springframework/security/web/authentication/switchuser/SwitchUserFilter.java]: Not a problem.

As an FYI...We try to have an assertion for any setter on a property that has a default value. There may be some older setters that don't follow this convention. Also if the default value of the property is null, then there is no need for the assertion.

**Pull request closed without merging**